aten/src/ATen/native/utils/ParamsHash.h

```diff
@@ -38,5 +38,59 @@ struct ParamsEqual {
   }
 };
 
+// Provide explicit byte-for-byte constructors to avoid uwittingly leaving
+// padding bytes unitialized (e.g., when passing Params by value)
+template <typename T>
+struct ParamsWrapper {
+  T pod;
+  static_assert(std::is_standard_layout_v<T>, "ParamsWrapper cannot wrap non-POD data");
```

Comment:
I don't know if it matters if `pod` is actually POD or just `standard_layout`, but you could check `std::is_pod`